rb_tree.py

```diff
@@ -35,29 +37,34 @@ def __eq__(self, other):
         if self.parent is None or other.parent is None:
             parents_are_same = self.parent is None and other.parent is None
         else:
-            parents_are_same = self.parent.value == other.parent.value and self.parent.color == other.parent.color
-        return self.value == other.value and self.color == other.color and parents_are_same
+            parents_are_same = self.parent.value
+            self.parent.value == other.parent.value and self.parent.color
+            other.parent.value and self.parent.color == other.parent.color
+        return (self.value == other.value and
+                self.color == other.color and parents_are_same)
 
     def has_children(self) -> bool:
-        """ Returns a boolean indicating if the node has children """
```

Comment:
Why remove this comment?

Reply:
Actually I don't realized that it was removed. I'll put all these comments in their places again.